packages/mds-ingest-service/repository/index.ts

```diff
@@ -472,20 +472,30 @@ class IngestReadWriteRepository extends ReadWriteRepository {
 
   public getLatestTelemetryForDevices = async (device_ids: UUID[]): Promise<TelemetryDomainModel[]> => {
     try {
-      const connection = await this.connect('ro')
-      const entities = await connection
-        .createQueryBuilder(TelemetryEntity, 'telemetry')
-        .innerJoin(
-          subquery =>
-            subquery
-              .select('id, RANK() OVER (PARTITION BY device_id ORDER BY timestamp DESC) AS rownum')
-              .where('device_id = ANY(:device_ids)', { device_ids })
-              .from(TelemetryEntity, 't'),
-          'last_device_telemetry',
-          'last_device_telemetry.id = telemetry.id AND last_device_telemetry.rownum = 1'
-        )
-        .getMany()
-      return entities.map(TelemetryEntityToDomain.mapper())
+      /**
+       * https://github.com/typeorm/typeorm/blob/master/docs/transactions.md#using-queryrunner-to-create-and-control-state-of-single-database-connection
+       */
+      const runner = await (await this.connect('ro')).createQueryRunner()
```

Comment:
I’ve never loved `await await`. Could we just call it a connection? :-)

Reply:
Yes! This is what I was _hoping_ to find when i was reading typeORM api docs. Less try/catches, less verbose. Thanks!